sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableWriteIT.java

```diff
@@ -0,0 +1,190 @@
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * End-to-end tests of BigtableWrite.
+ */
+@RunWith(JUnit4.class)
+public class BigtableWriteIT implements Serializable {
+  /**
+   * These tests requires a static instances because the writers go through a serialization step
+   * when executing the test and would not affect passed-in objects otherwise.
+   */
+  private static final String COLUMN_FAMILY_NAME = "cf";
+  private static BigtableTestOptions options;
+  private BigtableOptions bigtableOptions;
+  private static BigtableSession session;
+  private static BigtableTableAdminClient tableAdminClient;
+
+  @Before
+  public void setup() throws Exception {
+    PipelineOptionsFactory.register(BigtableTestOptions.class);
+    options = TestPipeline.testingPipelineOptions().as(BigtableTestOptions.class);
+
+    BigtableOptions.Builder bigtableOptionsBuilder = new BigtableOptions.Builder()
+        .setProjectId(options.getProjectId())
+        .setClusterId(options.getClusterId())
+        .setZoneId(options.getZoneId())
+        .setUserAgent("apache-beam-test");
+    bigtableOptions = bigtableOptionsBuilder.build();
+
+    session = new BigtableSession(bigtableOptions);
+    tableAdminClient = session.getTableAdminClient();
+  }
+
+  @Test
+  public void testE2EBigtableWrite() throws Exception {
+    final String tableId = String.format("BigtableWriteIT-%tF-%<tH-%<tM-%<tS-%<tL", new Date());
+    final String tableName = bigtableOptions.getClusterName().toTableNameStr(tableId);
+    final String clusterName = bigtableOptions.getClusterName().toString();
+    final int numRows = 1000;
+    final List<KV<ByteString, ByteString>> testData = generateTableData(numRows);
+
+    createEmptyTable(clusterName, tableId);
+
+    Pipeline p = Pipeline.create(options);
+    p.apply(CountingInput.upTo(numRows))
+        .apply(ParDo.of(new DoFn<Long, KV<ByteString, Iterable<Mutation>>>() {
+          @Override
+          public void processElement(ProcessContext c) {
+            int index = c.element().intValue();
+
+            Iterable<Mutation> mutations =
+                ImmutableList.of(Mutation.newBuilder()
+                    .setSetCell(
+                        Mutation.SetCell.newBuilder()
+                            .setValue(testData.get(index).getValue())
+                            .setFamilyName(COLUMN_FAMILY_NAME))
+                    .build());
+            c.output(KV.of(testData.get(index).getKey(), mutations));
+          }
+        }))
+        .apply(BigtableIO.write()
+          .withBigtableOptions(bigtableOptions)
+          .withTableId(tableId));
+    p.run();
+
+    // Test number of column families and column family name equality
+    Table table = getTable(tableName);
+    assertEquals(table.getColumnFamilies().size(), 1);
```

Comment:
`assertEquals("Column family size", 1, table.getColumnFamilies().size());`

Two things going on here:

1. Add an error message, so that when it fails we don't just see: "expected 1: got: 0" -- we see a little more context also.
2. Put expected first in `assertEquals`.